ext/rugged/rugged.h

```diff
@@ -147,6 +147,12 @@ struct rugged_cb_payload
     int exception;
 };
 
+struct rugged_treecount_cb_payload
+{
+	unsigned int count;
+	unsigned int limit;
```

Comment:
You're using `-1` as a "no limit" signifier below, so either this has to go back to a signed type (because otherwise `payload->limit >= 0` will always be true) or you can special case `0` instead.

Personally, I would probably make these both `ssize_t` values. If `ssize_t` isn't easily usable from within Rugged for portability reasons, you could make both simple `int` values, too.